File: LEETCODE/LEETCODE-CPP/387.cpp
//387. First Unique Character in a String

//method 1
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char,int> m;
        int n=s.length();
        
        for(int i=0;i<n;i++)
            m[s[i]]++;
        
        for(int i=0;i<n;i++)
            if(m[s[i]]==1)
                return i;
        
        return -1;
    }
};


//method 2
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> fq(26,0);
        
        for(int i=0;i<s.length();i++)
            fq[s[i]-'a']++;
        
        for(int i=0;i<s.length();i++)
            if(fq[s[i]-'a']==1)
                return i;
        
        return -1;
    }
};

File: LEETCODE/LEETCODE-JAVA/387.java
//387. First Unique Character in a String
//https://leetcode.com/problems/first-unique-character-in-a-string/

//Method-1 using hashmap
class Solution {
    public int firstUniqChar(String s) {
        char st[]=s.toCharArray();
        int n=s.length();

        HashMap<Character,Integer> m= new HashMap<>();

        for(char c:st){
            m.put(c,m.getOrDefault(c,0)+1);
        }

        for(int i=0;i<n;i++){
            if(m.get(st[i])==1){
                return i;
            }
        }

        return -1;
    }
}